File: medplum-java/medplum-java-model/src/main/java/com/fhir/Plandefinition_actionSelectionbehavior.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class Plandefinition_actionSelectionbehavior {
  public static final Plandefinition_actionSelectionbehavior ANY = new Plandefinition_actionSelectionbehavior(Value.ANY, "ANY");

  public static final Plandefinition_actionSelectionbehavior ALL = new Plandefinition_actionSelectionbehavior(Value.ALL, "ALL");

  private final Value value;

  private final String string;

  Plandefinition_actionSelectionbehavior(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof Plandefinition_actionSelectionbehavior && this.string.equals(((Plandefinition_actionSelectionbehavior) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case ANY:
        return visitor.visitANY();
      case ALL:
        return visitor.visitALL();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static Plandefinition_actionSelectionbehavior valueOf(@Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "ANY":
        return ANY;
      case "ALL":
        return ALL;
      default:
        return new Plandefinition_actionSelectionbehavior(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    ANY,

    ALL,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitANY();

    T visitALL();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/Plandefinition_actionSelectionbehavior.ts
export type Plandefinition_actionSelectionbehavior =
    | Plandefinition_actionSelectionbehavior.Any
    | Plandefinition_actionSelectionbehavior.All;

export const Plandefinition_actionSelectionbehavior = {
    Any: "any" as Plandefinition_actionSelectionbehavior.Any,
    All: "all" as Plandefinition_actionSelectionbehavior.All,

    _visit: <Result>(
        value: Plandefinition_actionSelectionbehavior,
        visitor: Plandefinition_actionSelectionbehavior._Visitor<Result>
    ): Result => {
        switch (value) {
            case Plandefinition_actionSelectionbehavior.Any:
                return visitor.any();
            case Plandefinition_actionSelectionbehavior.All:
                return visitor.all();
            default:
                return visitor._unknown();
        }
    },
};

export declare namespace Plandefinition_actionSelectionbehavior {
    export type Any = "any" & {
        __Any: void;
    };
    export type All = "all" & {
        __All: void;
    };

    export interface _Visitor<Result> {
        any: () => Result;
        all: () => Result;
        _unknown: () => Result;
    }
}
